nncf/common/graph/graph.py

```diff
@@ -294,9 +434,130 @@ def topological_sort(self) -> List[NNCFNode]:
                                                     key=lambda x: self._nx_graph.nodes[x][NNCFGraph.ID_NODE_ATTR])]
 
     def dump_graph(self, path: str):
-        """
-        Writes graph in .dot format.
-
-        :param path: Path to save.
-        """
-        nx.drawing.nx_pydot.write_dot(self._nx_graph, path)
+        nx.drawing.nx_pydot.write_dot(self.get_graph_for_structure_analysis(), path)
+
+    def visualize_graph(self, path: str):
+        out_graph = self._get_graph_for_visualization()
+        nx.drawing.nx_pydot.write_dot(out_graph, path)
+        try:
+            A = to_agraph(out_graph)
+            A.layout('dot')
+            png_path = os.path.splitext(path)[0]+'.png'
+            A.draw(png_path)
+        except ImportError:
+            nncf_logger.warning("Graphviz is not installed - only the .dot model visualization format will be used. "
+                                "Install pygraphviz into your Python environment and graphviz system-wide to enable "
+                                "PNG rendering.")
+
+    def get_graph_for_structure_analysis(self, extended=False) -> nx.DiGraph:
+        """The graph to dump has certain node attributes omitted, compared to the graph stored
+         inside NNCFGraph."""
+        out_graph = nx.DiGraph()
+        for node_name, node in self._nx_graph.nodes.items():
+            attrs_node = {
+                'id': node[NNCFGraph.ID_NODE_ATTR],
+                'type': node[NNCFGraph.NODE_TYPE_ATTR]
+            }
+            if 'color' in node:
+                attrs_node['color'] = node['color']
+            if 'label' in node:
+                attrs_node['label'] = node['label']
+            if 'style' in node:
+                attrs_node['style'] = node['style']
+
+            out_graph.add_node(node_name, **attrs_node)
+        if extended:
+            for u, v in self._nx_graph.edges:
+                out_graph.add_edge(u, v, label=self._nx_graph.edges[u, v][NNCFGraph.ACTIVATION_SHAPE_EDGE_ATTR])
+        else:
+            for u, v in self._nx_graph.edges:
+                out_graph.add_edge(u, v)
+
+        return out_graph
+
+    def _get_graph_for_visualization(self) -> nx.DiGraph:
+        """A user-friendly graph .dot file, making it easier to debug the network and setup
```

Comment:
Please, format the docstring according to the Style Guide.

Reply:
Done